src/main.rs

```diff
@@ -317,7 +322,7 @@ fn main() {
                     .to_ascii_uppercase();
 
                     println!("\n{}: {}", genes[selection], descs[selection]);
-                    print_seq(&s);
+                    print_seq(&s).unwrap();
```

Comment:
I had to call `.unwrap()` here because `print_seq` now returns a `Result<(), Error>`, but `unwrap`s are generally harder to debug when they happen because there is no context on the error message. An alternative is using `.expect("reason why the error occured")`, which will still panic and abort the program, but will have a more meaningful error message.

(The even better way is actually checking for the error and exiting properly, like you had with `process::exit(1)`. Left as an exercise to the reader =])